crates/procedurals/src/lib.rs

```diff
@@ -0,0 +1,192 @@
+use std::{collections::HashMap, sync::Arc};
+
+use ambient_core::{
+    asset_cache,
+    bounding::{local_bounding_aabb, world_bounding_aabb, world_bounding_sphere},
+    main_scene,
+    transform::{local_to_world, mesh_to_world},
+};
+use ambient_ecs::{
+    components,
+    generated::components::core::procedurals::{procedural_material, procedural_mesh},
+    query, Entity, Resource, SystemGroup,
+};
+use ambient_gpu::{mesh_buffer::GpuMesh, texture::TextureView};
+use ambient_renderer::{
+    gpu_primitives_lod, gpu_primitives_mesh,
+    pbr_material::{get_pbr_shader, PbrMaterial, PbrMaterialConfig},
+    primitives, renderer_shader, SharedMaterial,
+};
+use ambient_shared_types::{
+    ProceduralMaterialHandle, ProceduralMeshHandle, ProceduralSamplerHandle,
+    ProceduralTextureHandle,
+};
+use ambient_std::{cb, mesh::Mesh};
+
+components!("procedurals", {
+    @[Resource]
+    procedural_storage: ProceduralStorage,
+});
+
+pub fn systems() -> SystemGroup {
```

Comment:
suggest `client_systems` to make it obvious this only runs on the client for now